src/GafferOSL/OSLShader.cpp

```diff
@@ -319,6 +333,20 @@ Plug *loadCompoundNumericParameter( const OSLQuery::Parameter *parameter, const
 	/// \todo Get from metadata
 	ValueType minValue( Imath::limits<BaseType>::min() );
 	ValueType maxValue( Imath::limits<BaseType>::max() );
+	if( metadata )
+	{
+		typedef typename TypedParameter<ValueType>::ObjectType ValueDataType;
```

Comment:
Any reason we can't use `<TypedData<ValueType>>` here? I'm very reluctant to introduce a dependency on `IECore::Parameter`, because I'm aiming to get all dependencies on that out of Gaffer, so that Gaffer can depend on only a slimmed down version (or subset) of Cortex in the future.